deepsensor/data/loader.py

```diff
@@ -1413,30 +1460,87 @@ def sample_sliding_window(
             stride = patch_size
         
         dy, dx = stride
-
         # Calculate the global bounds of context and target set.
         x1_min, x1_max, x2_min, x2_max = self.coord_bounds
-
         ## start with first patch top left hand corner at x1_min, x2_min
         patch_list = []
 
-        for y in np.arange(x1_min, x1_max, dy):
-            for x in np.arange(x2_min, x2_max, dx):
-                if y + x1_extend > x1_max:
-                    y0 = x1_max - x1_extend
-                else:
-                    y0 = y
-                if x + x2_extend > x2_max:
-                    x0 = x2_max - x2_extend
-                else:
-                    x0 = x
+        # Todo: simplify these elif statements
+        if self.coord_directions['x1'] == False and self.coord_directions['x2'] == True:
+            for y in np.arange(x1_max, x1_min, -dy):
```

Comment:
```
            for y in np.arange(x1_max, x1_min, -dy):
                    y0 = x1_max - x1_extend
                for x in np.arange(x2_min, x2_max, dx):
                else:
                    if y - x1_extend < x1_min:
                    y0 = y
                        y0 = x1_min + x1_extend
                if x + x2_extend > x2_max:
                    else:
                    x0 = x2_max - x2_extend
                        y0 = y
                else:
                    if x + x2_extend > x2_max:
                    x0 = x
                        x0 = x2_max - x2_extend
                    else:
                        x0 = x

                    # bbox of x1_min, x1_max, x2_min, x2_max per patch
                    bbox = [y0 - x1_extend, y0, x0, x0 + x2_extend]
                    patch_list.append(bbox)
```

This sort of block is repeated  4 times here, could we turn this into a reusable function?

Reply:
Btw I'm keen to get this PR merged so if this is a blocker, we could come back to it.